### PR TITLE
Mag timeseries types

### PR DESCRIPTION
**Pull request by znicholls**
# Pull request

Please confirm that this pull request has done the following:

- [x] Re-wrote sample MAG files to match conventions settled on here (and any related docs)
- [x] Merged https://github.com/lewisjared/scmdata/pull/9
- [x] Tests added
- [x] Documentation added (where applicable)
- [x] Example added (either to an existing notebook or as a new notebook, where applicable) (N/A)
- [x] Description in ``CHANGELOG.rst`` added

## Adding to CHANGELOG.rst

Please add a single line in the changelog notes similar to one of the following:

```
- (`#XX <http://link-to-pr.com>`_) Added feature which does something
- (`#XX <http://link-to-pr.com>`_) Fixed bug identified in (`#XX <http://link-to-issue.com>`_)
```

### CODE REVIEW

**Comment by znicholls** [comment]
@rgieseke what is the best way to document this?

File: pymagicc/magicc_time.py [hunk 1]
```diff
@@ -173,3 +173,40 @@ def _yr_frac_close_to(yfrac, other, must_be_greater=False):
...
+    if ttype in ("POINT_MID_YEAR", "AVERAGE_YEAR_MID_YEAR"):
+        df.index = df.index.map(lambda x: dt.datetime(x, 7, 1))
+        return df
+
+    if ttype in ("POINT_END_YEAR", "AVERAGE_YEAR_END_YEAR"):
+        df.index = df.index.map(lambda x: dt.datetime(x, 12, 31))
+        return df
+
+    if ttype in ("MONTHLY",):
+        df.index = df.index.map(convert_to_datetime)
```
[the remark this answers is not in the document]
Comment (lewisjared) [reply pymagicc/magicc_time.py]: Or depending on the month

[the remark this answers is not in the document]
Comment (znicholls) [reply pymagicc/magicc_time.py]: That is an option. At the moment the time column in a monthly file has float times so it just reads those out and then converts those to datetimes. It means you get all this extra minute info which is a bit misleading so could be a change. What do you think?

[the remark this answers is not in the document]
Comment (znicholls) [reply pymagicc/magicc_time.py]: After talking with @lewisjared, I'll change this to make it so the datetimes come back in a more sensible way (which reflects their intent better).

[the remark this answers is not in the document]
Comment (znicholls) [reply pymagicc/magicc_time.py]: It looks like this is already how `convert_to_datetime` works so I think we're ok here.

**Comment by znicholls** [comment]
@rgieseke are these docs enough?

**Comment by znicholls** [comment]
@rgieseke can I merge this?

**Comment by rgieseke** [comment]
> @rgieseke can I merge this?

Not anymore, i already did.
Sorry, i hadn't seen the notification.

**Comment by znicholls** [comment]
Beauty thanks!

**Pull request merged**